source/addonStore/models/addon.py

```diff
@@ -161,13 +166,21 @@ def isPendingInstall(self) -> bool:
 			# have not been installed yet
 			addonDataManager._downloadsPendingInstall,
 		)
+		nameInDownloadsPendingCompletion = filter(
+			lambda m: m.model.name == self.name,
+			# add-ons which are currently being downloaded
+			# and have not been cancelled
+			addonDataManager._downloadsPendingCompletion,
+		)
 		return (
 			super().isPendingInstall
 			# True if this add-on has been downloaded but
 			# has not been installed yet
 			or bool(next(nameInDownloadsPendingInstall, False))
 			# True if this add-on is currently being downloaded
+			# and the download has not been cancelled
 			or os.path.exists(self.tempDownloadPath)
+			and bool(next(nameInDownloadsPendingCompletion, False))
```

Comment:
Is there a situation where `os.path.exists(self.tempDownloadPath)` is true and `bool(next(nameInDownloadsPendingCompletion, False))` is False? Should we just use this check instead?

```suggestion
			and bool(next(nameInDownloadsPendingCompletion, False))
```

Alternatively, can you add brackets to clarify the logic more clearly

```suggestion
			or (
					os.path.exists(self.tempDownloadPath)
					and bool(next(nameInDownloadsPendingCompletion, False))
			)
```